src/sagas/background-refresh.js

```diff
@@ -325,33 +328,76 @@ function* scheduleResumingNotifications ({ payload: { delayInSeconds } }) {
   }
 }
 
-/**
- * When ovirt-web-ui is installed to ovirt-engine, a logout should push the user to the
- * base ovirt welcome page.  But when running in dev mode or via container, the logout
- * page is displayed.  In that case, we want to make sure the page is set to something
- * appropriate and that background refreshing is no longer done.
- */
-function* logoutAndCancelScheduler () {
-  yield put(Actions.setCurrentPage({ type: C.NO_REFRESH_TYPE }))
-  yield put(Actions.stopSchedulerFixedDelay())
-  yield put(Actions.stopSchedulerForResumingNotifications())
+//
+// ** BACKGROUND_REFRESH channel handler **
+//
+function* handleBackgroundChannel (action) {
+  const { subType } = action
+
+  // NOTE: The timer is self-cancelling by using `BACKGROUND_REFRESH` as a cancel action
+  let startNewTimer = true
+  let timerDuration =
+    yield select(({ options }) => options.getIn(['remoteOptions', 'refreshInterval', 'content'], AppConfiguration.schedulerFixedDelayInSeconds))
+
+  switch (subType) {
+    case 'changePage':
+      yield refreshDataForCurrentPage({ pageRouterRefresh: true })
+      break
+
+    case 'manual':
+      yield refreshDataForCurrentPage({ manualRefresh: true })
+      break
+
+    case 'timer':
+      yield refreshDataForCurrentPage({ schedulerRefresh: true })
+      break
+
+    case 'fetchNextPageOfVmsAndPools':
+      yield fetchByPage()
+      break
+
+    case 'restartBackgroundRefreshTimer':
+      timerDuration = yield calculateStartDelayFromLastRefresh()
+      break
+
+    case 'stop':
+      startNewTimer = false
+      break
+  }
+
+  if (startNewTimer) {
+    yield spawn(backgroundRefreshTimer, timerDuration)
+  }
 }
 
-export default [
-  // only process the 1st manual refresh received in a 5 second window
-  throttle(5000, C.MANUAL_REFRESH, refreshManually),
-
-  /*
-   * Note: If a user goes crazy swapping between pages very quickly, a lot of `CHANGE_PAGE`
-   *       actions will be fired.  Since changePage() puts `START_SCHEDULER_FIXED_DELAY`
-   *       to refresh the new page data each time, it is possible to get to a point where
-   *       multiple page refresh sagas are running at the same time.  That can cause
-   *       problems and slow down the app.  Not much to be done to prevent this from
-   *       happening without slowing down the responsiveness of the app.
-   */
-  takeLatest(C.CHANGE_PAGE, changePage),
-
-  takeEvery(C.START_SCHEDULER_FIXED_DELAY, startSchedulerWithFixedDelay),
-  takeEvery(C.START_SCHEDULER_FOR_RESUMING_NOTIFICATIONS, scheduleResumingNotifications),
-  takeEvery(C.LOGOUT, logoutAndCancelScheduler),
-]
+export default function* () {
+  const backgroundChannel = yield actionChannel(BACKGROUND_REFRESH)
+
+  return [
+    takeEvery(backgroundChannel, handleBackgroundChannel),
```

Comment:
> but this line introduces again race conditions between them

I wasn't able to generate any, but maybe?  The alternative to make sure things run right is to have another saga that does a `take()`/`call()` loop instead of the `take()`/`fork()` loop.  That should be an easy enough switch.

Reply:
New impl will make sure the serialized logic flows smoothly

Reply:
Test scenario:
1. configure custom network filter in dev tools with 5 sec delay
2. load VMs List screen
3. 4-5 times trigger infinite scroller by scrolling up&down
4. click manual refresh

### Before

![image](https://user-images.githubusercontent.com/64194103/124445365-c6c02b80-dd7f-11eb-9b73-ee583f353421.png)

Results:
1. all request overlap
2. the same page is fetched by infinite scroller

### After

![image](https://user-images.githubusercontent.com/64194103/124444922-5f09e080-dd7f-11eb-80b8-cce799d34c81.png)

Results:
1. requests are serialized
2. different page is fetched  by infinite scroller

Reply:
The after is what I was going for....